### PR TITLE
ADMINAPI-1077 Add Get/Post endpoints for healthchecks

### PR DESCRIPTION
**Pull request by dfernandez-gap**

### CODE REVIEW

**Review by jagudelo-gap — CHANGES_REQUESTED** [review]
@DavidJGapCR @jleiva-gap @dfernandez-gap @ricardogamboagap 
We should include some e2e, just a simple validation of the endpoints. If it is complicated, we have to create a ticket for that.

We have to consider how to include or execute the scripts in the database automatically (if flag is enabled), maybe the runtime migration can help us by the time being, but it doesn't seem to be the best approach based on Microsoft recommendation. I would prefer scripts but you could find a better approach.

https://learn.microsoft.com/en-us/ef/core/managing-schemas/migrations/applying?tabs=dotnet-core-cli#apply-migrations-at-runtime

File: Application/EdFi.Ods.AdminApi/Program.cs [hunk 1]
```diff
@@ -1,24 +1,38 @@
...
+//Rate Limit
+builder.Services.AddMemoryCache();
+builder.Services.Configure<IpRateLimitOptions>(builder.Configuration.GetSection("IpRateLimiting"));
+builder.Services.AddSingleton<IIpPolicyStore, MemoryCacheIpPolicyStore>();
+builder.Services.AddSingleton<IRateLimitCounterStore, MemoryCacheRateLimitCounterStore>();
+builder.Services.AddSingleton<IRateLimitConfiguration, RateLimitConfiguration>();
 builder.Services.AddInMemoryRateLimiting();
-// logging
+
+if (builder.Configuration.GetValue<bool>("AppSettings:EnableAdminConsoleAPI"))
```
[the remark this answers is not in the document]
Comment (dfernandez-gap) [reply Application/EdFi.Ods.AdminApi/Program.cs]: Added change as requested

File: Application/EdFi.Ods.AdminApi/Program.cs [hunk 2]
```diff
@@ -1,24 +1,38 @@
...
+{
+    ServiceRegistration.AddRepositories(builder.Services);
```
[the remark this answers is not in the document]
Comment (dfernandez-gap) [reply Application/EdFi.Ods.AdminApi/Program.cs]: Added change as requested

File: Application/EdFi.Ods.AdminApi.AdminConsole/Infrastructure/Services/HealthChecks/Queries/GetHealthCheckQuery.cs [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    private readonly IQueriesRepository<HealthCheck> _healthCheckQuery;
+
+    public GetHealthCheckQuery(IQueriesRepository<HealthCheck> healthCheckQuery)
+    {
+        _healthCheckQuery = healthCheckQuery;
+    }
+    public async Task<HealthCheck> Execute(int tenantId)
+    {
+        return await _healthCheckQuery.Query().SingleOrDefaultAsync(healthCheck => healthCheck.TenantId == tenantId)
+        ?? throw new Exception($"Not found {nameof(HealthCheck)} for Tenant Id: {tenantId}");
```
[the remark this answers is not in the document]
Comment (dfernandez-gap) [reply Application/EdFi.Ods.AdminApi.AdminConsole/Infrastructure/Services/HealthChecks/Queries/GetHealthCheckQuery.cs]: Added change as requested

[the remark this answers is not in the document]
Comment (DavidJGapCR) [reply Application/EdFi.Ods.AdminApi.AdminConsole/Infrastructure/Services/HealthChecks/Queries/GetHealthCheckQuery.cs]: https://edfi.atlassian.net/browse/ADMINAPI-1084
Added a ticket to cover e2e tests.

**Review by jagudelo-gap — APPROVED** [review]

**Pull request merged**